dali/pipeline/operators/color/hsv.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/color/hsv.h"
+#include "dali/kernels/imgproc/pointwise/linear_transformation_cpu.h"
+
+namespace dali {
+namespace {
+
+template <typename Out, typename In>
+using TheKernel = kernels::LinearTransformationCpu<Out, In, 3, 3, 3>;
+}  // namespace
+
+namespace hsv {
+
+const std::string kHue = "hue";                 // NOLINT
+const std::string kSaturation = "saturation";   // NOLINT
+const std::string kValue = "value";             // NOLINT
+const std::string kOutputType = "output_type";  // NOLINT
+
+}  // namespace hsv
+
+
+DALI_SCHEMA(Hsv)
+              .DocStr(R"code(This operator performs HSV manipulation.
+To change hue, saturation and/or value of the image, pass corresponding coefficients.
+Keep in mind, that `hue` has additive delta argument,
+while for `saturation` and `value` they are multiplicative.
+
+This operator accepts RGB color space as an input.
+
+HSV manipulation, for performance reasons, is conducted by transition
+to YIQ color space.
```

Comment:
I wouldn't explicitly name YIQ color space here - but, since we're not actually using HSV, I'd mention something like:
```For performance reasons, the operation is approximated by a linear transform in RGB space.
The color vector is projected along the neutral (gray) axis, rotated (according to hue delta) and scaled according to value and saturation multiplers, and then restored to original color space.
```
The exact formula or commitment to a particular color space would impose backward compatibility - even, if we find a way to improve the function.

Reply:
Done